JANGAN/ProjectTools/BaseKerasModelTrainer.py

```
@@ -0,0 +1,137 @@
+    Model = None
+    Datasets = []
+    Epochs = 0
+    CurrentEpoch = None
+    RefreshUIEachXStep = 1
+    SaveCheckpoints = False
+    CheckpointPath = ""
+    LatestCheckpointPath = ""
+
+    def __init__(self, model, datasets, epochs, refreshUIEachXStep, saveCheckPoints, checkpointPath, latestCheckpointPath):
+        self.Model = model
+        self.Datasets = datasets
+        self.Epochs = epochs
+        self.RefreshUIEachXStep = refreshUIEachXStep
+        self.SaveCheckpoints = saveCheckPoints
+        self.CheckpointPath = checkpointPath
+        self.LatestCheckpointPath = latestCheckpointPath
+
+    def TrainModel(self):
+        print("Training started")
+        for epoch in range(self.Epochs):
+            self.CurrentEpoch = epoch
+            start = time.time()
+
+            print(f"Epoch {epoch + 1} of {self.Epochs} is in progress...")
+            
+            self.EpochRun(epoch)
+
+            totalEpochTime = time.time()-start
+
+            print(f"Time for epoch {epoch + 1} is {self.GetDatetimeFromSeconds(totalEpochTime)}. Est time remaining for training is {self.GetDatetimeFromSeconds(totalEpochTime*(self.Epochs-(epoch + 1)))}")
+        
+        print("Training finished!")
+            
+    def CreateDataSet(self, dataArray):
+        (returnTrainSet, returnTestSet) = dataArray[0]
+        for data in dataArray[1:]:
+            (addTrainSet, addTestSet) = data
+            returnTrainSet = returnTrainSet.concatenate(addTrainSet)
+            returnTestSet = returnTestSet.concatenate(addTestSet)
+        
+        returnTrainSet = returnTrainSet.shuffle(buffer_size=1024)
+        returnTestSet = returnTestSet.shuffle(buffer_size=1024)
+        return (returnTrainSet, returnTestSet)
+
+    def GetDatetimeFromSeconds(self, seconds):
+        return time.strftime("%H:%M:%S", time.gmtime(seconds))
+
+    def PrintStatus(self, iteration, totalIterations, epochTime, epoch):
+        print("No Train loss text implemented.", end="\r")
```

Is it not better to throw an exception?